### DISCUSSION TITLE
Retrieving Entity & Wikipedia URL from Knowledge Base

### DISCUSSION THREAD

**Discussion by amitbcp** [explosion/spaCy#5366] [Help: Coding & Implementations] [answered]
I created a KB using the bni/wiki_entity_linking example.
I have two doubts which I think would be possible but is missing from documentation.

1.  Example kb.get_candidates("Adam") : Returns Candidates in descending order of prior probability. Is it possible to fetch the entity_value for the QID ? Like for the Q<ID> the Entity is Adam XX & Alias is Adam ?

2. Since we are using it for NEL. How to I get the wikipedia page for the entity/candidate ? i.e. Q<ID> refers to this URL ??

## Your Environment

* Operating System: RHEL 7
* Python Version Used: 3.6.5
* spaCy Version Used: 2.2.4

**Comment by svlandeg** [comment, marked as answer]
> 1. Example kb.get_candidates("Adam") : Returns Candidates in descending order of prior probability. Is it possible to fetch the entity_value for the QID ? Like for the Q the Entity is Adam XX & Alias is Adam ?

I'm not sure what you mean by "entity_value". If you link the QID to the wikipedia page (cf answer to question 2), then you'll get the full name (wikipedia title) of the entity. You can also check the intermediary file `entity_alias.csv` that has been generated and which has all the recorded aliases:

```
cat entity_alias.csv

WD_id|alias
Q31|Kingdom of Belgium
Q31|be
Q31|🇧🇪
Q8|😄
Q8|:)
Q8|😃
Q8|joy
Q8|happy
...
```


> 2. Since we are using it for NEL. How to I get the wikipedia page for the entity/candidate ? i.e. Q refers to this URL ??
> 

The NEL approach here uses Wikidata's Q identifiers as the unique ID to which to resolve. This is done because Wikidata identifiers are much more stable than Wikipedia page titles. However if you've run the scripts, you should have a file `entity_defs.csv` which has a mapping between the two:

```
cat entity_defs.csv

WP_title|WD_id
Belgium|Q31
Happiness|Q8
George Washington|Q23
Jack Bauer|Q24
Douglas Adams|Q42
Paul Otlet|Q1868
Portugal|Q45
Antarctica|Q51
...
```

**Comment by amitbcp** [comment]
@svlandeg  thank you so much for the quick response.This really helps.
